src/components/AvatarWithIndicator.js

```diff
@@ -14,37 +19,116 @@ const propTypes = {
     // avatar size
     size: PropTypes.string,
 
+    // Whether we show the sync indicator
+    isSyncing: PropTypes.bool,
 };
 
 const defaultProps = {
     isActive: false,
     size: 'default',
+    isSyncing: false,
 };
 
-const AvatarWithIndicator = ({
-    isActive,
-    source,
-    size,
-}) => {
-    const indicatorStyles = [
-        size === 'large' ? styles.statusIndicatorLarge : styles.statusIndicator,
-        isActive ? styles.statusIndicatorOnline : styles.statusIndicatorOffline,
-    ];
-
-    return (
-        <View
-            style={[size === 'large' ? styles.avatarLarge : styles.sidebarAvatar]}
-        >
-            <Avatar
-                style={[size === 'large' ? styles.avatarLarge : null]}
-                source={source}
-            />
-            <View style={StyleSheet.flatten(indicatorStyles)} />
-        </View>
-    );
-};
+class AvatarWithIndicator extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.rotate = new Animated.Value(0);
+        this.scale = new Animated.Value(1);
+        this.startSyncIndicator = this.startSyncIndicator.bind(this);
+        this.stopSyncIndicator = this.stopSyncIndicator.bind(this);
+    }
+
+    componentDidMount() {
+        if (this.props.isSyncing) {
+            this.startSyncIndicator();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.isSyncing && this.props.isSyncing) {
+            this.startSyncIndicator();
+        } else if (prevProps.isSyncing && !this.props.isSyncing) {
+            this.stopSyncIndicator();
+        }
+    }
+
+    componentWillUnmount() {
+        this.stopSyncIndicator();
+    }
+
+    startSyncIndicator() {
+        Animated.loop(Animated.timing(this.rotate, {
+            toValue: 1,
+            duration: 3000,
+            easing: Easing.linear,
+            isInteraction: false,
+            useNativeDriver: true,
+        })).start();
+        Animated.spring(this.scale, {
+            toValue: 1.35,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start();
+    }
+
+    stopSyncIndicator() {
+        Animated.spring(this.scale, {
+            toValue: 1,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start(() => {
+            this.rotate.resetAnimation();
+            this.scale.resetAnimation();
+            this.rotate.setValue(0);
+        });
+    }
+
+    render() {
+        const indicatorStyles = [
+            styles.alignItemsCenter,
+            styles.justifyContentCenter,
+            {
```

Comment:
We've tried to take a pretty hard stance against inline styles. In this scenario, we would usually make a function in `styles.js` called `getAvatarIndicatorStyles` or something and dynamically generate these styles in there. You can see how I handed transforms in a dynamic style function in `src/styles/getTooltipSyles.js`. We haven't really decided whether dynamic style generators should be in their own module or directly in `styles.js`, so it's up to you! Since it's probably just one function, I would probably put it directly in `styles.js`